src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int sizeOfList;
+    private Node firstNode;
+    private Node lastNode;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node node = new Node(value);
+        if (sizeOfList == 0) {
+            firstNode = node;
+            lastNode = node;
+            sizeOfList += 1;
+            return;
+        }
+        lastNode.nextNode = node;
+        node.previousNode = lastNode;
+        lastNode = node;
+        sizeOfList += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == sizeOfList) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            addToStart(value, index);
+            return;
+        }
+        Node currentNextNode = getNodeByIndex(index);
+        Node currentPrevNode = currentNextNode.previousNode;
+        Node node = new Node(value);
+        currentPrevNode.nextNode = node;
+        node.previousNode = currentPrevNode;
+        node.nextNode = currentNextNode;
+        currentNextNode.previousNode = node;
+        sizeOfList += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node currentNode = getNodeByIndex(index);
+        return (T) currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node currentNode = getNodeByIndex(index);
+        T currentValue = (T) currentNode.value;
+        currentNode.value = value;
+        return currentValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node deletedNode = getNodeByIndex(index);
```

Comment:
include index check insider of retrieval method, instead of calling them one after another every time